### PR TITLE
Fixed VAULT_TOKEN expand in order to be able to access it in TF

### PR DESCRIPTION
**Pull request by mvisonneau**

### CODE REVIEW

File: terraform.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ fi
 if [ -n "${VAULT_ADDR}" ]; then
     if [ -z "${VAULT_TOKEN}" ]; then
         if [ -n "${VAULT_ROLE_ID}" ] && [ -n "${VAULT_SECRET_ID}" ]; then
-            export VAULT_TOKEN=$(curl -s -X POST -d "{\"role_id\":\"${VAULT_ROLE_ID}\",\"secret_id\":\"${VAULT_SECRET_ID}\"}" "${VAULT_ADDR}/v1/auth/approle/login" | jq -r .auth.client_token)
+            declare "VAULT_TOKEN"=$(curl -s -X POST -d "{\"role_id\":\"${VAULT_ROLE_ID}\",\"secret_id\":\"${VAULT_SECRET_ID}\"}" "${VAULT_ADDR}/v1/auth/approle/login" | jq -r .auth.client_token)
```
Comment (paulRbr) [inline terraform.sh]: Don't you need `declare -x` to export the variable?

Comment (mvisonneau) [reply terraform.sh]: ah right! let me test locally see how it goes

**Comment by mvisonneau** [comment]
I've also replaced the fetch of the creds with the a non deprecated method : GET instead of POST

**Comment by paulRbr** [comment]
Is it backwards compatible (the `GET`) ?

**Comment by mvisonneau** [comment]
yep, thanks 🙏 !

**Pull request merged**